packages/core/core/src/Transformation.js

```diff
@@ -264,15 +264,7 @@ export default class Transformation {
       initialCacheEntry || (await this.runPipeline(pipeline, initialAsset));
 
     if (!initialCacheEntry) {
-      let resultCacheKey = this.getCacheKey(
-        [initialAsset],
-        pipeline.configs,
-        await getInvalidationHash(
-          assets.flatMap(asset => asset.getInvalidations()),
-          this.options,
-        ),
-      );
-      await this.writeToCache(resultCacheKey, assets, pipeline.configs);
+      await this.writeToCache(initialAssetCacheKey, assets, pipeline.configs);
```

Comment:
I believe this is intentional. The invalidations can't be known before transforming, so the cache key that we write to is different than the original cache key we check if on an initial build. On subsequent builds, we send the invalidations from the graph as part of the request so hopefully the cache key should map. Perhaps the problem is that whatever is in the graph is not matching?